backend/src/impl/analyses/significance_analysis.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+from explainaboard.info import SysOutputInfo
+from explainaboard.metrics.metric import MetricStats
+from explainaboard.metrics.registry import get_metric_config_class
+from explainaboard.utils.typing_utils import unwrap
+
+
+def update_metric_config(system_output: SysOutputInfo) -> SysOutputInfo:
+    metric_configs = [
+        get_metric_config_class(metric_config_dict["cls_name"])(**metric_config_dict)
+        for metric_config_dict in system_output.metric_configs
+    ]
+    # update metric_configs in system_output
+    system_output.metric_configs = metric_configs
+
+    return system_output
+
+
+"""
+significance test based on bootstrapped resampling method, which are
+controlled by two major
+hyper-parameters:
+(1) n_samples: The number of bootstrapped samples
+(2) prop_samples: The ratio of samples to take every time
+
+"""
+
+
+def significance_test(
+    sys1_info: SysOutputInfo,
+    sys2_info: SysOutputInfo,
+    sys1_metric_stats: MetricStats,
+    sys2_metric_stats: MetricStats,
+    n_samples: int = 1000,
+    prop_samples: float = 0.5,
+) -> dict:
+
+    sys1_metric_names = []
+    sys2_metric_names = []
+    metric_funcs = []
+    sig_info = {}
+
+    for config in unwrap(sys1_info.analysis_levels[0].metric_configs):
+        sys1_metric_names.append(config.name)
+        metric_funcs.append(config.to_metric())
+    for config in unwrap(sys2_info.analysis_levels[0].metric_configs):
+        sys2_metric_names.append(config.name)
+
+    # metric validation
+    if sys1_metric_names != sys2_metric_names:
+        raise ValueError(
+            f"Evaluation metrics of two system should be with the same "
+            f"name and same order: {sys1_metric_names},"
+            f"{sys2_metric_names}"
+        )
+
+    for sys1_metric_stat, sys2_metric_stat, metric_func, metric_name in zip(
+        sys1_metric_stats, sys2_metric_stats, metric_funcs, sys1_metric_names
+    ):
+
+        n_elems = max(int(prop_samples * len(sys1_metric_stat)), 1)
+        all_indices = np.array(range(len(sys1_metric_stat)))
+        rng = np.random.default_rng()
```

Comment:
I think this `np.random.default_rng()` call can be moved outside the for loop. This function call actually takes non-trivial time.